classes/local/rule_instances_controller.php

```diff
@@ -168,18 +160,61 @@ public function get_rule_instance_records(): array {
     }
 
     /**
-     * Return the rule instance record matching the given instanceid.
+     * Return a list of hydrated rule instance objects.
+     *
+     * @return rule\rule_interface[]|rule\plugin_configurable[]|rule\instance_configurable[]
```

Comment:
My only remark left here: I think `get_rule_instances()`-method's PHPDoc should only mention `rule\rule_interface[]` as return type as it's our base minimum for rule instance classes. As we've also defined in `get_rule_instance_by_id()`.

```suggestion
     * @return rule\rule_interface[]
```